File: pr-4510/sphinx/examples/python/operators.py
# [Begin Spin]
import cudaq
from cudaq import spin
# [End Spin]

# [Begin Pauli]
words = ['XYZ', 'IXX']
coefficients = [0.432, 0.324]


@cudaq.kernel
def kernel(coefficients: list[float], words: list[cudaq.pauli_word]):
    q = cudaq.qvector(3)

    for i in range(len(coefficients)):
        exp_pauli(coefficients[i], q, words[i])


# [End Pauli]

File: pr-4734/sphinx/examples/cpp/operators.cpp
/*******************************************************************************
 * Copyright (c) 2025 - 2026 NVIDIA Corporation & Affiliates.                  *
 * All rights reserved.                                                        *
 *                                                                             *
 * This source code and the accompanying materials are made available under    *
 * the terms of the Apache License 2.0 which accompanies this distribution.    *
 ******************************************************************************/

// [Begin Spin]
#include <cudaq.h>
// [End Spin]

// [Begin Pauli]
__qpu__ void kernel() {
  cudaq::qvector qvector(3);
  exp_pauli(0.432, qvector, "XYZ");
  exp_pauli(0.324, qvector, "IXX");
}

int main() {
  auto result = cudaq::sample(kernel);
  result.dump();
  return 0;
}
// [End Pauli]
